hermes/bin/src/runtime_extensions/hermes/http_gateway/event.rs

```diff
@@ -60,14 +203,137 @@ impl HermesEventPayload for HTTPEvent {
             &self.method,
         )?;
 
-        if let Some(resp) = event_response {
-            Ok(self.sender.send(HTTPEventMsg::HttpEventResponse((
-                resp.code,
-                resp.headers,
-                resp.body,
-            )))?)
-        } else {
-            Ok(())
+        match event_response {
+            Some(HttpGatewayResponse::Http(resp)) => {
+                self.send_http_response(resp.code, resp.headers, resp.body)
+            },
+            Some(HttpGatewayResponse::InternalRedirect(location)) => {
+                self.handle_internal_redirect(location)
+            },
+            None => Ok(()),
+        }
+    }
+}
+
+impl HTTPEvent {
+    /// Send HTTP response back to client via MPSC channel
+    fn send_http_response(&self, code: Code, headers: HeadersKV, body: Body) -> anyhow::Result<()> {
+        Ok(self
+            .sender
+            .send(HTTPEventMsg::HttpEventResponse((code, headers, body)))?)
+    }
+
+    /// Handle internal redirect with security validation
+    fn handle_internal_redirect(&self, location: String) -> anyhow::Result<()> {
+        let config = RedirectConfig::from_env();
+
+        if let Err(e) = validate_redirect_location(&location, &config) {
+            error!("Invalid redirect location: {}", e);
+            return self.send_error_response(403, "Forbidden: Invalid redirect location");
+        }
+
+        self.spawn_redirect_request(location);
+        Ok(())
+    }
+
+    /// Spawn background thread for redirect request
+    fn spawn_redirect_request(&self, location: String) {
+        let headers = self.headers.clone();
+        let method = self.method.clone();
+        let body = self.body.clone();
+        let sender = self.sender.clone();
+
+        std::thread::spawn(move || {
```

Comment:
We might consider issuing a spike to investigate if it is feasible to allow async in the event handlers, so we can use tokio task instead of OS thread.

Reply:
yes I agree, it would improve performance.. I couldn't find a clean way of doing it so kept it simple for now.